### ISSUE TITLE
Can't select/get from context when using `NestFactory.createMicroservice`

### ISSUE THREAD

**Issue by tmaciej** [nestjs/nest#398]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
Components from imported modules cannot be accessed through context/container from microservices. Methods `select` and `get` do not exist on `INestMicroservice` (`NestMicroservice` class does not extend `NestApplicationContext`).

```ts
// This works
const app = await NestFactory.create(SomeModule);
app.select(SomeImportedModule).get(SomeComponent);

// This does not
const app = await NestFactory.createMicroservice(SomeModule, {
    // ...options
});
app.select(SomeImportedModule).get(SomeComponent); // methods select and get does not exist on microservice instance
```

## Expected behavior
Components from imported modules should be accessible from microservice context.

## Minimal reproduction of the problem with instructions
1. Create a module that imports another module
2. Create microservice using previously created module
3. Try to get component instance from imported module

(Example code above)

## What is the motivation / use case for changing the behavior?
Using global filters / pipes / interceptors / guards etc. from external (imported) modules should be possible.

## Environment

<pre><code>
Nest version: 4.5.7

 
For Tooling issues:
- Node version: 8.9.4
- Platform: Mac

Others:
-
</code></pre>

**Comment by kamilmysliwiec** [comment]
Thanks for reporting, I'm on it

**Comment by kamilmysliwiec** [comment]
It's fixed in `v4.6.0` 🙂

**Comment by tmaciej** [comment]
@kamilmysliwiec could you possibly provide an example how to do it correctly? I upgraded to `v4.6.2` and get following error when calling `select` on microservice instance

```
[Nest] 31   - 2018-2-16 10:16:14   [ExceptionHandler] Cannot read property 'metatype' of null
TypeError: Cannot read property 'metatype' of null
     at NestMicroservice.select (/usr/app/node_modules/@nestjs/core/nest-application-context.js:14:51)
     at exceptions_zone_1.ExceptionsZone.run (/usr/app/node_modules/@nestjs/core/nest-factory.js:115:48)
     at Function.run (/usr/app/node_modules/@nestjs/core/errors/exceptions-zone.js:16:13)
     at Proxy.args (/usr/app/node_modules/@nestjs/core/nest-factory.js:114:54)
     at /usr/app/dist/index.js:24:39
     at Generator.next (<anonymous>)
     at fulfilled (/usr/app/dist/index.js:4:58)
     at <anonymous>
     at process._tickCallback (internal/process/next_tick.js:188:7)
     at Function.Module.runMain (module.js:686:11)
```

Code example (excerpt):

```ts
async function bootstrap() {
    const app = await NestFactory.createMicroservice(TestModule, {
        // ...
    });

    const loggerInterceptor = app.select(LoggerModule).get(LoggerRPCInterceptor);
    app.useGlobalInterceptors(loggerInterceptor);
}
```

```ts
@Module({
    // ...
    imports: [
        LoggerModule,
    ],
    // ...
})
export class TestModule { }
```

```ts
// Logger module
@Module({
    // ...
    components: [
        // ...
        LoggerRPCInterceptor,
        // ...
    ],
    exports: [
        // ...
        LoggerRPCInterceptor
        // ...
    ],
})
export class LoggerModule {
}
```
